scipy/optimize/_numdiff.py

```diff
@@ -516,7 +516,19 @@ def _dense_difference(fun, x0, f0, h, use_one_sided, method):
         else:
             raise RuntimeError("Never be here.")
 
-        J_transposed[i] = df / dx
+        if np.count_nonzero(dx == 0.0):
+            # this can happen for entries with lb == ub
+            with np.errstate(invalid='ignore'):
+                # filter warning for 0 / 0
+                # However a 1.0 / 0.0 will still raise a
+                # RuntimeWarning: divide by zero encountered in true_divide
+                g = np.asarray(df / dx)
+            idxs = np.logical_and(np.array(df) == 0, np.array(dx) == 0)
```

Comment:
This overhead should only be experienced by the ill posed problems where lb == ub